eva_submission/submission_qc_checks.py

```diff
@@ -0,0 +1,316 @@
+from retry import retry
+
+from eva_submission.eload_submission import Eload
+from eva_submission.submission_config import EloadConfig
+
+logger = logging_config.get_logger(__name__)
+
+
+class EloadQC(Eload):
+    def __init__(self, eload_number, config_object: EloadConfig = None):
+        super().__init__(eload_number, config_object)
+        self.profile = cfg['maven']['environment']
+        self.private_config_xml_file = cfg['maven']['settings_file']
+        self.project_accession = self.eload_cfg.query('brokering', 'ena', 'PROJECT')
+        self.path_to_data_dir = Path(cfg['projects_dir'], self.project_accession)
+        self.taxonomy = self.eload_cfg.query('submission', 'taxonomy_id')
+        self.analyses = self.eload_cfg.query('brokering', 'analyses')
+
+    def check_if_study_appears_in_dev(self):
+        dev_url = f"https://wwwdev.ebi.ac.uk/eva/webservices/rest/v1/studies/{self.project_accession}/summary"
+        json_response = self.get_result_from_webservice(dev_url)
+        if self.check_if_study_present_in_response(json_response, 'id', self.project_accession):
+            self._study_dev_check_result = "PASS"
+        else:
+            self._study_dev_check_result = "FAIL"
+
+        return f"""
+                pass: {self._study_dev_check_result}"""
+
+    def check_if_study_appears_in_variant_browser(self, species_name):
+        dev_url = f"https://wwwdev.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/list?species={species_name}"
+        json_response = self.get_result_from_webservice(dev_url)
+        if self.check_if_study_present_in_response(json_response, 'studyId', self.project_accession):
+            return True
+        else:
+            return False
+
+    def check_if_study_appears_in_dev_metadata(self):
+        missing_assemblies = []
+        for analysis_data in self.analyses.values():
+            species_name = self.get_species_name(analysis_data['assembly_accession'])
+            if not self.check_if_study_appears_in_variant_browser(species_name):
+                missing_assemblies.append(f"{species_name}({analysis_data['assembly_accession']})")
+
+        self._study_dev_metadata_check_result = "PASS" if not missing_assemblies else "FAIL"
+        return f"""
+                pass: {self._study_dev_metadata_check_result}
+                missing assemblies: {missing_assemblies if missing_assemblies else None}"""
+
+    @retry(tries=3, delay=2, backoff=1.5, jitter=(1, 3))
+    def get_result_from_webservice(self, url):
+        response = requests.get(url)
+        response.raise_for_status()
+        return response.json()
+
+    def check_if_study_present_in_response(self, res, key, project_accession):
+        if any(res) and 'response' in res and len(res['response']) > 0:
+            for response in res['response']:
+                if response['numTotalResults'] >= 1:
+                    for result in response['result']:
+                        if result[key] == project_accession:
```

Comment:
```suggestion
                        if result[key] == self.project_accession:
```
(& remove the method param)

Reply:
done